server/src/main/java/org/opensearch/rest/extensions/RestSendToExtensionAction.java

```diff
@@ -104,12 +105,20 @@ public RestSendToExtensionAction(
                 if (parts.length > 2) {
                     name = Optional.of(parts[2].trim());
                 }
+                if (parts.length > 3) {
+                    legacyActionName = Optional.of(parts[3].trim());
+                }
             } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
                 throw new IllegalArgumentException(restAction + " does not begin with a valid REST method");
             }
             logger.info("Registering: " + method + " " + path);
             if (name.isPresent()) {
```

Comment:
Lets not allow for anything but named routes to be used by extensions.  Since they are net new this isn't a breaking change.

Reply:
This was implemented to support plugins transforming to extensions. Without this the only way to support legacyActionNames is via privilege evaluation in security plugin. By implementing this, we're allowing plugins to register transport action names for each route. @cwperks your thoughts on this?

Reply:
I agree with @peternied that extensions should be required to use NamedRoutes. That will break the SDK in its current form so let's make sure to have a PR ready in the SDK for the breaking change. I first implemented this as optional to prevent it breaking the SDK and planned to follow it up when it was more concrete to enforce that extensions use NamedRoutes.